src/box/lua/config/init.lua

```diff
@@ -128,6 +130,10 @@ function methods._initialize(self)
     if extras ~= nil then
         extras.initialize(self)
     end
+
+    self:_register_source(require('internal.config.source.env').new({
```

Comment:
For me, it looks more correct to have a module for each source.

Reply:
The idea is to write the logic of collecting environment variable values once: `<schema>:pairs()` call and the postprocessing. It reduces a chance to change one source and forget to change another one.

There is another option: implement a kind of 'base class' with this logic and two small modules. I considered it and it seems a bit too complex for me.

Is there other option to don't repeat the env vars collection logic?

TBH, I think that using an object table as a storage for internal values is the good practice (as opposite to using module's upvalues). At least, it allows introspection: we can look inside the internal state of the object.

I appreciate that you marked it as non-blocking comment, but I'm open for suggestions/discussion here.